script/checks/evm-stack-addresses.ts

```diff
@@ -0,0 +1,89 @@
+import { type Address, type Hex, type PublicClient, keccak256 } from 'viem';
+
+export type EVMStack = 'OP' | 'Orbit';
+
+type StackAccount = {
+	name: string;
+	address: Address;
+	kind: 'Predeploy' | 'Precompile';
+};
+
+export type EVMStackResult = StackAccount & {
+	codeHash: Hex;
+	exists: boolean;
+};
+
+const NO_CODE_HASH = keccak256('0x');
+const INVALID_CODE_HASH = keccak256('0xfe');
+
+export async function checkEvmStackAddresses(
+	client: PublicClient,
+): Promise<Record<EVMStack, EVMStackResult[]>> {
+	const result: Record<EVMStack, EVMStackResult[]> = {
+		OP: [],
+		Orbit: [],
+	};
+
+	for (const stack of Object.keys(evmStackAddresses)) {
+		const stackPredploys = evmStackAddresses[stack as EVMStack];
+		const res = stackPredploys.map(async ({ name, address, kind }) => {
+			const code = await client.getBytecode({ address });
```

Comment:
Would it make sense to do `getBytecode({ address, blockNumber: 0n })` here? Or can predeploys be added at a later block (e.g., a hardfork)?

Reply:
Not super important right now because the current addresses are obviously not deployable. The problem is that checking for "nothing-under-my-sleeve"ness seems tricky.

Reply:
Nvm, some of these are added at later blocks, like `0x4200000000000000000000000000000000000042`.

I think it's still _technically_ possible to automatically check that they weren't deployed but it's involved as hell and not worth it.

Reply:
> The problem is that checking for "nothing-under-my-sleeve"ness seems tricky.

You are right that checking for this is very hard—even if the block number was 0 (i.e. it was deployed at genesis as expecteed) there still could be malicious state set (in that predeploy, or elsewhere) at genesis.

> Nvm, some of these are added at later blocks, like `0x4200000000000000000000000000000000000042`.

For other OP Stack chains this address will have been deployed at genesis. For OP Mainnet this address existed pre-bedrock when predeploys were placed differently, which is why there is a deployment transaction.